app/assets/javascripts/discourse/tests/acceptance/topic-list-tracker-test.js

```diff
@@ -3,11 +3,45 @@ import {
   previousTopicUrl,
   setTopicId,
 } from "discourse/lib/topic-list-tracker";
-import { acceptance } from "discourse/tests/helpers/qunit-helpers";
+import { acceptance, exists } from "discourse/tests/helpers/qunit-helpers";
 import { test } from "qunit";
-import { visit } from "@ember/test-helpers";
+import { click, visit } from "@ember/test-helpers";
+import topicFixtures from "discourse/tests/fixtures/topic";
+import discoveryFixtures from "discourse/tests/fixtures/discovery-fixtures";
+import { cloneJSON } from "discourse-common/lib/object";
+import { NotificationLevels } from "discourse/lib/notification-levels";
+
+acceptance("Topic list tracking", function (needs) {
+  let notificationLevel;
+
+  needs.hooks.afterEach(() => {
+    notificationLevel = null;
+  });
+
+  needs.user();
+
+  needs.pretender((server, helper) => {
+    server.get("/latest.json", () => {
+      const fixture = discoveryFixtures["/latest.json"];
+
+      if (notificationLevel) {
+        fixture["topic_list"]["topics"].find((t) => {
```

Comment:
I think we have to clone this _before_ changing the object/value, otherwise it will modify it everywhere

Reply:
Good catch! I missed this one here.